### PR TITLE
fix(Node SDK): Add SetTimeout() to Node SDK

### LINKED ISSUE #1250: [Bug Report] Node SDK is missing Sandbox.setTimeout() for updating sandbox idle TTL
## Summary

The Node SDK does not expose `Sandbox.setTimeout()`, preventing users from updating a sandbox's idle timeout (idle TTL) while it is running.

The server already implements `POST /sandboxes/{sandboxID}/timeout`; both the Python SDK (#743  ) and the Go SDK(#850 )provide corresponding wrappers. Only the Node SDK lacks this capability.

## Environment

- CubeSandbox version / commit: latest `master`
- Host OS and kernel version: N/A, SDK-only issue
- KVM info (`modinfo kvm`): N/A, SDK-only issue
- Deployment mode: N/A, SDK-only issue
- Relevant component: other / Node SDK (`sdk/node`, `@cubesandbox/sdk` 0.3.0)

## Steps to Reproduce

1. Create a sandbox with the Node SDK:

   ```ts
   import { Sandbox } from "@cubesandbox/sdk";
   
   const sb = await Sandbox.create({ template: "tpl-test" });
   ```

2. Attempt to update the idle TTL while the sandbox is running:

   ```
   await sb.setTimeout(3600);
   ```

3. TypeScript compilation fails:

   ```
   error TS2339: Property 'setTimeout' does not exist on type 'Sandbox'.
   ```

## Expected Behavior

The Node SDK should expose the following method on `Sandbox` instances:

```
async setTimeout(timeout: number): Promise<void>
```

The method should call the already supported control-plane endpoint:

```
POST /sandboxes/{sandboxID}/timeout
Content-Type: application/json
```

Expected request body:

```
{
  "timeout": 3600
}
```

Expected `timeout` semantics:

```
await sb.setTimeout(3600);          // sends timeout: 3600
await sb.setTimeout(0.5);           // 500ms, sends timeout: 1
await sb.setTimeout(0);             // expires immediately
await sb.setTimeout(NEVER_TIMEOUT); // -1, disables idle timeout
```

The method should accept the server's `204 No Content` success response and reuse the SDK's existing error mapping: a 404 response should throw `SandboxNotFoundError`, while other non-success responses should map to the existing API error types.

## Actual Behavior

The Node SDK's `Sandbox` class has no `setTimeout()` method, and the package entry point does not export an equivalent `NEVER_TIMEOUT` constant.

Users must bypass the SDK and hand-roll the control-plane HTTP request:

```
await fetch(`${apiUrl}/sandboxes/${sb.sandboxId}/timeout`, {
  method: "POST",
  headers: {
    "Content-Type": "application/json",
    Authorization: `Bearer ${apiKey}`,
  },
  body: JSON.stringify({ timeout: 3600 }),
});
```

This requires users to handle authentication, request timeouts, response status codes, and error mapping themselves, defeating the purpose of the SDK wrapper.

```
TypeScript users encounter TS2339 at compile time;
JavaScript users can only discover that this capability is unavailable at runtime.
```

## Additional Context

The server route, OpenAPI contract, and backend implementation already exist. This is therefore a missing client-side wrapper in the Node SDK, not missing backend functionality.

The Python SDK already provides:

```
sb.set_timeout(3600)
```

The Go SDK already provides:

```
err := sb.SetTimeout(ctx, time.Hour)
```

The Go SDK accepts `time.Duration`; positive fractional seconds are rounded up to whole seconds before being sent to the server. The Node SDK should follow the same semantics.

Suggested regression tests:

```
- setTimeout(positive integer) sends the correct POST path and JSON request body.
- setTimeout(0) preserves timeout: 0.
- setTimeout(0.5) sends timeout: 1.
- setTimeout(NEVER_TIMEOUT) sends timeout: -1 and verifies that the constant can be imported from the package entry point.
- -2, NaN, and Infinity are rejected before a request is sent.
- A server 404 response maps to SandboxNotFoundError.
```

### PR DESCRIPTION
**Pull request by ywxzm03**
## Summary

Fix the missing `Sandbox.setTimeout()` API in the Node SDK, allowing users to update a sandbox's idle TTL while it is running.

Closes #1250

### Modified Files

- `sdk/node/src/config.ts` — Add the `NEVER_TIMEOUT = -1` constant.
- `sdk/node/src/index.ts` — Export `NEVER_TIMEOUT` from the Node SDK package entry point.
- `sdk/node/src/sandbox.ts` — Add `Sandbox.setTimeout(timeout)`, which calls `POST /sandboxes/:id/timeout`.
- `sdk/node/test/sandbox.test.ts` — Add regression coverage for timeout updates, boundary values, input validation, and error mapping.
- `sdk/node/README.md` — Document `sb.setTimeout(timeout)` in the API method table.

## API

```ts
async setTimeout(timeout: number): Promise<void>
```

## Behavior

- Positive integers update the sandbox idle TTL in seconds.
- `0` requests immediate expiry.
- `NEVER_TIMEOUT` (`-1`) disables the idle timeout.
- Positive fractional seconds are rounded up to whole seconds, matching the Go SDK's `time.Duration` semantics: `0.5` seconds is sent as `1` second.
- Other negative values, `NaN`, and `Infinity` are rejected before a request is sent.
- Reuses the Node SDK's existing control-plane error mapping; a 404 response results in `SandboxNotFoundError`.

## Testing

```
cd sdk/node
npm test -- sandbox.test.ts
npm run typecheck
npm test
```

Test results:

- `npm test -- sandbox.test.ts`: passed, 45 tests.
- `npm run typecheck`: passed.
- `npm test`: passed, 186 tests passed and 1 integration test skipped.
- `git diff --check`: passed.

## Not in Scope

- No changes to CubeAPI, CubeMaster, Cubelet, or lifecycle services.
- No changes to existing sandbox URL path escaping behavior.
- Does not expose `POST /sandboxes/:id/refreshes`.

### CODE REVIEW

File: sdk/node/src/sandbox.ts [hunk 1]
```diff
@@ -583,6 +583,28 @@ export class Sandbox {
...
+  /** POST /sandboxes/:id/timeout — set sandbox idle timeout. */
+  async setTimeout(timeout: number): Promise<void> {
+    if (!Number.isFinite(timeout)) {
+      throw new Error(`timeout must be a finite number of seconds, got ${timeout}`);
+    }
+    if (timeout < 0 && timeout !== NEVER_TIMEOUT) {
+      throw new Error(`timeout must be >= 0 or NEVER_TIMEOUT (-1), got ${timeout}`);
+    }
+
+    const seconds = Math.ceil(timeout);
```
[the remark this answers is not in the document]
Comment (ywxzm03) [reply sdk/node/src/sandbox.ts:595]: > The client-side validation rejects non-finite and negative-but-not-`-1` values, but there's no upper bound. The backend field `SetTimeoutRequest.timeout` is `i32` (max `2,147,483,647`); a value like `5_000_000_000` passes `Number.isFinite` + `>= 0` here, gets `Math.ceil`'d, and is sent verbatim. Serde then fails to deserialize it into `i32`, so the caller gets an opaque Axum 400/422 `ApiError` instead of this method's own clear validation error.客户端验证会拒绝非有限值以及非-1的负值，但没有上限。后端的 SetTimeoutRequest.timeout 字段类型是 i32（最大值 2,147,483,647）；类似 5_000_000_000 这样的值能通过这里的 Number.isFinite + >= 0 检查，经过 Math.ceil 处理后原样发送。随后 Serde 无法将其反序列化为 i32，导致调用方得到一个不透明的 Axum 400/422 ApiError，而不是此方法自身清晰的验证错误。
> 
> Consider capping at `0x7FFFFFFF` (or documenting the range), consistent with the defensive validation this method already applies to the other boundary cases. Note the Go/Python SDKs share this gap, so it's a robustness suggestion rather than a regression.建议将上限设为0x7FFFFFFF（或记录该范围），与此方法已应用于其他边界情况的防御性校验保持一致。请注意，Go/Python SDK也存在这一缺口，因此这属于健壮性建议，而非回归问题。

May I need to address the "non-blocking" issue mentioned by it?
